pkg/informer/listwatch.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright 2020 Red Hat, Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package informer
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/tools/cache"
+)
+
+// ListWatchMetrics stores the pointers of list/watch counter metrics.
+type ListWatchMetrics struct {
+	ListTotal  *prometheus.CounterVec
+	WatchTotal *prometheus.CounterVec
+}
+
+// NewListWatchMetrics takes in a prometheus registry and initializes and
+// registers the kube_events_exporter_list_total and
+// kube_events_exporter_watch_total metrics.
+// It returns those registered metrics.
+func NewListWatchMetrics(registry *prometheus.Registry) *ListWatchMetrics {
+	metrics := &ListWatchMetrics{
+		ListTotal: prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Name: "kube_events_exporter_list_total",
```

Comment:
I would like you to have a look at the following two patterns:
- kube_events_exporter_list_total of result and resource in the labels
vs 
- kube_events_exporter_list_total kube_events_exporter_errors_list_total of the result being a seperate metric and resource still being in the label.

There are some philosophies on this that would be good for you to read. Feel free to ask me or @simonpasquier if you can't find anything on this, he has made the contribution to prom-operator where he is using the later one.

Reply:
Indeed, I found some resources in that regard and thus decided to go with:
- kube_events_exporter_list_total
- kube_events_exporter_list_failed_total
- kube_events_exporter_watch_total
- kube_events_exporter_watch_failed_total

While I was at it, I also removed the resource label as I did not find it very useful in my particular case.

Reply:
Sounds good, is it ready for another review?

Reply:
Yes it is please have a new look at it